experimental/PropertyDDS/packages/property-properties/src/properties/primitive_type_casts.js

```diff
@@ -41,120 +41,120 @@ var _simpleCastFunctor = function (in_array, in_value) {
  * @alias property-properties._castFunctors
  */
 var _castFunctors = {
-  /**
-   * Casts the input value to a Uint64
-   * @param {number} in_value - The value to use in the cast
-   * @param {number} [in_radix = 10] An integer between 2 and 36 that represents the
-   *    radix (the base in mathematical numeral systems) of the above in_value if it is a string.
-   * @return {number} The casted value
-   * @protected
-   */
-  Uint64: function (in_value, in_radix) {
-    if (in_value instanceof Uint64) {
-      return in_value;
+    /**
+     * Casts the input value to a Uint64
+     * @param {number} in_value - The value to use in the cast
+     * @param {number} [in_radix = 10] An integer between 2 and 36 that represents the
+     *    radix (the base in mathematical numeral systems) of the above in_value if it is a string.
+     * @return {number} The casted value
+     * @protected
+     */
+    Uint64: function (in_value, in_radix) {
+        if (in_value instanceof Uint64) {
+            return in_value;
+        }
+        if (typeof in_value === 'string') {
+            return Uint64.fromString(in_value, in_radix);
+        }
+        if (typeof in_value === 'number') {
+            return Uint64.fromString(String(in_value), in_radix);
+        }
+        throw new Error(MSG.INT_64_NON_INT64_TYPE + ' , value: ' + in_value);
+    },
```

Comment:
Just FYI - @PaulKwMicrosoft just pointed out to me that browsers now have [BigInt](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt), [BigInt64Array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt64Array) and [BigUint64Array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigUint64Array) support.